clang/lib/Basic/Targets/AArch64.cpp

```diff
@@ -679,6 +679,7 @@ bool AArch64TargetInfo::hasFeature(StringRef Feature) const {
       .Case("f32mm", FPU & SveMode && HasMatmulFP32)
       .Case("f64mm", FPU & SveMode && HasMatmulFP64)
       .Case("sve2", FPU & SveMode && HasSVE2)
+      .Case("sve2p1", HasSVE2p1)
```

Comment:
It seems the changes for `HasSVE2p1` in this patch are not required for the tests to pass. I think the `hasFeature` method returns the value set by the Clang command rather than whatever is on the attribute. The `evaluateRequiredTargetFeatures` gets its information from elsewhere (not using this interface).

Reply:
Yep, you're right. I've removed these changes now.